### PR TITLE
Fix roundoff issue in SUNDIALS evolve()

### PR DESCRIPTION
**Pull request by ajnonaka**
## Summary
Fix a roundoff issue in the SUNDIALS evolve() function that caused the integrator to run a second time with a very tiny dt in a given time step.

## Checklist

The proposed changes:
- [X ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Src/Base/AMReX_RKIntegrator.H [hunk 1]
```diff
@@ -251,7 +251,8 @@ public:
         for (int step_number = 0; step_number < BaseT::max_steps && !stop; ++step_number)
         {
             // Adjust step size to reach output time
-            if (time_out - time_current < dt) {
+            // protect against roundoff
+            if (std::abs(time_out - time_current - dt) <= 1.e5 * std::numeric_limits<amrex::Real>::epsilon() * std::abs(dt)) {
```
Comment (WeiqunZhang) [inline Src/Base/AMReX_RKIntegrator.H]: What if Real is float?

We could also do `amrex::almostEqual(time_current+dt, time_out)`. `almostEqual` has an optional argument that can be used to fine tune what is considered almost equal.

Comment (ajnonaka) [reply Src/Base/AMReX_RKIntegrator.H]: That sounds good, I'll test it.

Comment (ajnonaka) [reply Src/Base/AMReX_RKIntegrator.H]: @WeiqunZhang how is this?

**Comment by WeiqunZhang** [comment]
Any reasons why `std::abs` is needed? Also the last argument should be an int (e.g., `10`, not double like `10.`.

**Comment by ajnonaka** [comment]
@WeiqunZhang fixed

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**